src/app/modules/shared/components/details-fields/details-fields.component.spec.ts

```diff
@@ -730,6 +729,33 @@ describe('DetailsFieldsComponent', () => {
       expect(numberinISOFormat).toBe(expectedISOFormatNumbers[numberIndex]);
     });
   });
+
+  it('when the user selects technologies, set them in the variable selectedTechnologies', () => {
+    const techs = ['php', 'angular'];
+    component.onTechnologiesUpdated(techs);
+    expect(component.selectedTechnologies).toEqual(techs);
+  });
+
+  it('when the user does not select a project, display a warning message.', () => {
+    spyOn(toastrServiceStub, 'warning');
+    component.onclickFormAction(true);
+    expect(toastrServiceStub.warning).toHaveBeenCalled();
+  });
+
+  it('call the close method if opened equals true', () => {
+    const datepicker: any = { opened : true, open : () => ({}), close : () => ({}) };
+    spyOn(datepicker, 'close');
+    component.openOrCloseDatePicker(datepicker);
+    expect(datepicker.close).toHaveBeenCalled();
+  });
+
+  it('call the open method if opened equals false', () => {
+    const datepicker: any = { opened : false, open : () => ({}), close : () => ({}) };
+    spyOn(datepicker, 'open');
+    component.openOrCloseDatePicker(datepicker);
+    expect(datepicker.open).toHaveBeenCalled();
+  });
```

Comment:
These unit tests are duplicated in the file "input-date.component.spec.ts". You should test a functionality only once, keep the tests in the file where the date picker is more relevant.
https://github.com/ioet/time-tracker-ui/blob/513e48cd8a0219b336650d6d6441b2973d52221f/src/app/modules/shared/components/input-date/input-date.component.spec.ts#L40-L52

Reply:
Luis is right. You should take a look at the checks when doing a PR. They are shown like these:
![image](https://user-images.githubusercontent.com/46400773/151597322-639d5514-884a-4855-a368-3b979a51dddc.png)
Please also check the [SonarCloud report](https://sonarcloud.io/component_measures?id=ioet_time-tracker-ui&pullRequest=800&metric=new_duplicated_lines_density&view=list) of this PR to see what's the problem (Code duplication) and where it is located